src/replication/ng/operation/flow/stage.go

```diff
@@ -185,28 +186,28 @@ func filterResources(resources []*model.Resource, filters []*model.Filter) ([]*m
 // step 1: get the detail information for each of the source namespaces
 // step 2: if the destination namespace isn't specified in the policy, then the
 // same namespaces with the source will be returned. If it is specified, then
-// returns the specified one with the merged metadatas of all source namespaces
+// returns the specified one with the merged metadata of all source namespaces
 func assembleDestinationNamespaces(srcAdapter adp.Adapter, srcResources []*model.Resource, dstNamespace string) ([]*model.Namespace, error) {
-	namespaces := []*model.Namespace{}
-	for _, srcResource := range srcResources {
-		namespace, err := srcAdapter.GetNamespace(srcResource.Metadata.Namespace)
-		if err != nil {
-			return nil, err
-		}
-		namespaces = append(namespaces, namespace)
-	}
-
 	if len(dstNamespace) != 0 {
-		namespaces = []*model.Namespace{
+		return []*model.Namespace{
```

Comment:
We have to set the metadata of destination namespace according to the metadata of source namespaces, so we must get the source namespaces first. And I have add a TODO in line 195.